File: 0169-majority-element/0169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
     const counter = {};
  for (let i = 0; i < nums.length; i++) {
    if (!(nums[i] in counter)) {
      counter[nums[i]] = 0;
    }
    counter[nums[i]] = counter[nums[i]] + 1;
  }
  let majorityCount = 0;
  let majorityItem;
  for (let num in counter) {
    if (counter[num] > majorityCount) {
      majorityCount = counter[num];
      majorityItem = num;
    }
  }
  return majorityItem;
};

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: list[int]) -> int:
        map = {}
        for i,v in enumerate(nums):
            if v in map.keys():
                map[v] += 1
            else:
                map[v] = 1
        for k,v in map.items():
            if(v > len(nums) / 2):
                return k